portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,24 +33,76 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> names;
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
-  public void init() {
-    names = new ArrayList<>();
-    names.add("Cindy");
-    names.add("Anthony");
-    names.add("Charles");
-    names.add("Chris");
-    names.add("Ben");
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment");
```

Comment:
The query parameters here ("comment" and "name") need to be spelled correctly both here and where used elsewhere. Likewise with the Datastore entity models and properties ("Comment", "comment-text", etc.).

Please add these as constants that you define once and reference elsewhere. This reduces the chances of introducing errors while resolving fields because the references are at least checked at compile time.

For example:
```
private static final String COMMENT_QUERY_PARAM = "comment";
private DatastoreService dataStore = ...
...
String comment = request.getParameter(COMMENT_QUERY_PARAM);
```